# 2.4. Problem Statement: STATISTICS 4


## Problem 1

Is gender independent of education level? A random sample of 395 people were surveyed and each person was asked to report the highest education level they obtained.<br/> The data that resulted from the survey is summarized in the following table:
|------------| High-School | Bachelor | Masters | Ph.D. | Total |
-----------------------------------------------------------------------------------
|Female--|60----------------|54-----------|46----------|41------|201---|
----------------------------------------------------------------------------------
|Male-----|40-----------------|44-----------|53----------|57------|194---|
-----------------------------------------------------------------------------------
|Total-----|100---------------|98-----------|99----------|98------|395---|
-----------------------------------------------------------------------------------
<br/>
Question: Are gender and education level dependent at 5% level of significance? In other words, given the data collected above, is there a relationship between the gender of an individual and the level of education that they have obtained?

## Solution

Since we are testing independence of two categorical variables (gender and education), we can use the Chi-square test of independence.<br/>
Here Null hypothesis (H0) is that the gender and education are independent. <br/>
Alternative Hypothesis(Ha) is that there is a dependency/relationship between gender and education

In [4]:
import numpy as np
from scipy import stats
male_array=[40,44,53,57]
female_array=[60,54,46,41]
fobs = np.array([female_array,male_array])
fobs

array([[60, 54, 46, 41],
       [40, 44, 53, 57]])

In [8]:
# Performing the chisquare test
ch2_stat, pvalue, dof, fexp = stats.chi2_contingency(fobs)
ch2_stat

8.006066246262538

In [12]:
print("chi-square statistic is:")
print(ch2_stat)
print()
print("p-value is:")
print(pvalue)
print()
print("Degrees of freedom is:")
print(dof)
print()
print("Expected values are:")
print(fexp)

chi-square statistic is:
8.006066246262538

p-value is:
0.045886500891747214

Degrees of freedom is:
3

Expected values are:
[[50.88607595 49.86835443 50.37721519 49.86835443]
 [49.11392405 48.13164557 48.62278481 48.13164557]]


In [13]:
print("The critical value for chi-square with 95% confidence and 3 dof is:")
crit = stats.chi2.ppf(0.95,3)
crit

The critical value for chi-square with 95% confidence and 3 dof is:


7.814727903251179

Since chi-square statistic 8.006 > critical value 7.814 , we can reject the null hypothesis and accept the alternative hypothesis. <br/>
So the conclusion is that there is a relationship between the gender of an individual and the level of education that they have obtained

## Problem 2

2. Using the following data, perform a one-way analysis of variance using α=.05. Write up the results in APA format. <br/>
[Group1: 51, 45, 33, 45, 67] <br/>
[Group2: 23, 43, 23, 43, 45] <br/>
[Group3: 56, 76, 74, 87, 56] <br/>

## Solution

Null Hypothesis H0: All groups have same mean. <br/>
Alternate Hypothesis Ha: The group means are all not equal, atleast one group has a different mean.<br/>

In [16]:
import numpy as np
from scipy import stats
grp1=[51,45,33,45,67]
grp2=[23,43,23,43,45]
grp3=[56,76,74,87,56]

In [18]:
fval,pval = stats.f_oneway(grp1,grp2,grp3)

In [20]:
print("F-value of the one way ANOVA is:",fval)
print("p-value of the F-distribution is:",pval)

F-value of the one way ANOVA is: 9.747205503009463
p-value of the F-distribution is: 0.0030597541434430556


In [59]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols 
df = pd.DataFrame({'grp1':grp1,'grp2':grp2,'grp3':grp3})
df['id']=df.reset_index()['index']+1
df_mod = pd.melt(df,['id'],['grp1','grp2','grp3'])
model = ols('value ~ variable', data=df_mod).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
anova_table

sum_sq    df         F   PR(>F)
variable  3022.933333   2.0  9.747206  0.00306
Residual  1860.800000  12.0       NaN      NaN

Since the p-value 0.003 < 0.05 We can reject the null hypothesis and accept the alternat hypothesis.<br/>
So we can conclude that atleast one group mean , among the three, is different from the other group means.<br/>

Further using F-test to test hypothesis.<br/>

In [64]:
from scipy.stats import f
F_critical = f.ppf(q=1-0.05,dfn=2,dfd=12)
F_critical

3.8852938346523933

Since F-test value (9.7472) > critical value(F_critical) for respective dof(2,12) we can reject the null hypothesis and accept the alternate hypothesis.<br/>
So we can conclude that atleast one group mean , among the three, is different from the other group means.

## Problem 3

Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25. <br/>
For 10, 20, 30, 40, 50:

## Solution

In [65]:
from statistics import stdev
arr1=[10,20,30,40,50]
arr2=[5,10,15,20,25]
VarA = stdev(arr1)**2
print("Variance of first array:",VarA)
VarB = stdev(arr2)**2
print("Variance of second array:",VarB)
ftest = VarA/VarB
print("F Test for given data:",ftest)

Variance of first array: 250.0
Variance of second array: 62.5
F Test for given data: 4.0
